stdlib/base.pkl

```diff
@@ -1367,6 +1367,12 @@ external class String extends Any {
   /// Splits this string around matches of [pattern].
   external function split(pattern: String|Regex): List<String>
 
+  /// Splits this string around the first [limit] matches of [pattern].
```

Comment:
Technically, around the first `limit-1` matches, no?

How much of a win (in what context/application) is this compared to `str.split(pat).take(limit)`?

Reply:
This came up in the context of inputs like `key=value=with=equals` where eg. you want to split `key` from the remainder after the first `=`. This is easy to accomplish like this:
```pkl
input = "key=value=with=equals"
delimiter = "="
key = input.split(delimiter).first
value = input.split(delimiter).drop(1).join(delimiter)
```

This gets significantly trickier when the split separator is a `Regex`, i.e. you must support both `key=value` and `key:value`
```pkl
input = "key=value=with=equals"
delimiter = Regex("[:=]")
key = input.split(delimiter).first
value = input.split(delimiter).drop(1).join(<???>) // <- no possible always-valid input here
```

You have to approach this differently, eg.
```pkl
input = "key=value=with=equals"
delimiter = Regex("[:=]")
key = input.split(delimiter).first
value = input.drop(input.indexOf(delimiter) + 1)
```

This requires reason about character indexes and match length. In this case the match is always a single character, but more complex cases (eg. the delimiter can be multiple or a variable number of characters) require additional handling. And—subjectively—is more difficult to read fluently.

With `splitLimit` this becomes much simpler to write and reason about:
```
input = "key=value=with=equals"
delimiter = Regex("[:=]")
key = input.split(delimiter).first
value = input.splitLimit(delimiter, 2).last
```

Reply:
Docstring updated!

Reply:
Ah, but in that case, I'd expect
```
  local str7 = "aabbccaabbccaabbccaabbcc"

   ["splitLimit()"] {
     str7.splitLimit(Regex("b+c"), 1)
```
to correspond to
```
["splitLimit()"] {
     List("aa", "caabbccaabbccaabbcc")
```
instead of the current
```
["splitLimit()"] {
     List("aa")
```

... but I must be misunderstanding something, because in your example, I would then expect
```
value = input.splitLimit(delimiter, 1).last
```
(if the `limit` counts the matches _of the separator_).

Reply:
Ignore that last bit; I hadn't absorbed

> If the limit n is greater than zero then the pattern will be applied at most n - 1 times, the array's length will be no greater than n, and the array's last entry will contain all input beyond the last matched delimiter.